tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -206,5 +243,70 @@ TEST(KNN, classify_multiple_kdtree)
 	SG_UNREF(features_test);
 	SG_UNREF(labels_test);
 }
-#endif /* HAVE_LAPACK */
 
+TEST(KNN, classify_multiple_kdtree)
+{
+	KNNTestData* mockdata = new KNNTestData();
+
+	CMulticlassLabels* labels = new CMulticlassLabels(mockdata->return_lab());
+	CDenseFeatures< float64_t >* features = new CDenseFeatures< float64_t >(mockdata->return_feat());
+
+	CFeatures* features_test = (CFeatures*) features->clone();
+	CLabels* labels_test = (CLabels*) labels->clone();
+
+	int32_t k=4;
+	CEuclideanDistance* distance = new CEuclideanDistance();	
+	CKNN* knn=new CKNN (k, distance, labels, KNN_KDTREE);
+	SG_REF(knn);
+
+	mockdata->set_up(labels, features, features_test, labels_test);
+
+	knn->train(features);
+
+	CEuclideanDistance* dist = new CEuclideanDistance(features, ((CDotFeatures*)features_test));	
+	knn->set_distance(dist);
+	SGMatrix<int32_t> out_mat =knn->classify_for_multiple_k();
+	features_test->remove_subset();
+
+	for ( index_t i = 0; i < labels_test->get_num_labels(); ++i )
+		for ( index_t j = 0; j < k; ++j )
+			EXPECT_EQ(out_mat(i, j), ((CMulticlassLabels*)labels_test)->get_label(i));
+
+	SG_UNREF(knn);
+	SG_UNREF(features_test);
+	SG_UNREF(labels_test);
+}
+
+TEST(KNN, classify_multiple_kcovertree)
+{
+	KNNTestData* mockdata = new KNNTestData();
+
+	CMulticlassLabels* labels = new CMulticlassLabels(mockdata->return_lab());
+	CDenseFeatures< float64_t >* features = new CDenseFeatures< float64_t >(mockdata->return_feat());
+
+	CFeatures* features_test = (CFeatures*) features->clone();
+	CLabels* labels_test = (CLabels*) labels->clone();
+
+	int32_t k=4;
+	CEuclideanDistance* distance = new CEuclideanDistance();	
+	CKNN* knn=new CKNN (k, distance, labels, KNN_COVER_TREE);
+	SG_REF(knn);
+
+	mockdata->set_up(labels, features, features_test, labels_test);
+
+	knn->train(features);
+
+	CEuclideanDistance* dist = new CEuclideanDistance(features, ((CDotFeatures*)features_test));	
+	knn->set_distance(dist);
+	SGMatrix<int32_t> out_mat =knn->classify_for_multiple_k();
+	features_test->remove_subset();
+
+	for ( index_t i = 0; i < labels_test->get_num_labels(); ++i )
+		for ( index_t j = 0; j < k; ++j )
+			EXPECT_EQ(out_mat(i, j), ((CMulticlassLabels*)labels_test)->get_label(i));
+
+	SG_UNREF(knn);
+	SG_UNREF(features_test);
+	SG_UNREF(labels_test);
+}
+#endif /* HAVE_LAPACK */
```

Comment:
Hmm, at least no related error and warning comes out after we remove the lapack macro.